src/devices/sound/nes_vt_apu.cpp

```diff
@@ -0,0 +1,289 @@
+// license:BSD-3-Clause
+// copyright-holders:David Shah
+/*****************************************************************************
+
+  MAME/MESS VTxx APU CORE
+
+ *****************************************************************************/
+
+#include "nes_vt_apu.h"
+#include "emu.h"
+
+#define VTXX_NTSC_XTAL XTAL(21'477'272)
+#define VTXX_NTSC_APU_CLOCK (VTXX_NTSC_XTAL/12)
+
+DEFINE_DEVICE_TYPE(NES_VT_APU, nesapu_vt_device, "nesapu_vt", "VTxx APU")
+DEFINE_DEVICE_TYPE(NES_VT_APU_SLAVE, nesapu_vt_slave_device, "nesapu_vt_slave", "VTxx APU (slave)")
+
+nesapu_vt_device::nesapu_vt_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock)
+	: nesapu_device(mconfig, tag, NES_VT_APU, owner, clock),
+	  m_xop2(*this, "nesapu_vt_slave"),
+	  m_rom_read_cb(*this)
+{
+
+}
+
+void nesapu_vt_device::device_start()
+{
+	apu_init();
```

Comment:
If you can't call the base class `device_start` it's usually a sign that the design is broken.  If there's stuff the regular NES APU does that the VT APU doesn't, that should be split into a derived class, so both APUs can use a common base.